lib/pure/httpclient.nim

```diff
@@ -150,6 +150,31 @@ proc code*(response: Response | AsyncResponse): HttpCode
   ## corresponding ``HttpCode``.
   return response.status[0 .. 2].parseInt.HttpCode
 
+proc contentType*(response: Response | AsyncResponse): string =
+  ## Retrieves the specified response's content type.
+  ## This is effectively the value of the "content-type" header.
+  ##
+  response.headers.getOrDefault("content-type")
+
+proc contentLength*(response: Response | AsyncResponse): int =
+  ## Retrieves the specified response's content length.
+  ## This is effectively the value of the "Content-Length" header.
+  ##
+  var contentLengthHeader = response.headers.getOrDefault("Content-Length")
+  discard contentLengthHeader.parseSaturatedNatural(result)
+
+proc lastModified*(response: Response | AsyncResponse): DateTime =
+  ## Retrieves the specified response's last modified time.
+  ## This is effectively the value of the "last-modified" header.
+  ## If the response's does not have a corresponding value
+  ## then "01 Jan 0000 00:00:00" will be returned.
```

Comment:
Meh, nowhere else do we use a "null" DateTime. This should just raise the ValueError that ``parse`` raises.

Reply:
Done.
Also I fixed date's parsing bug.
> Error: unhandled exception: index out of bounds [IndexError]

if token's value is empty.